pkg/injector/pod_patch.go

```diff
@@ -142,10 +146,57 @@ func (i *injector) getPodPatchOperations(ar *v1beta1.AdmissionReview,
 			Value: value,
 		},
 	)
+	patchOps = append(patchOps, envPatchOps...)
 
 	return patchOps, nil
 }
 
+func setDaprEnvVars(list []corev1.Container, basePath string) []PatchOperation {
+	envPatchOps := []PatchOperation{}
+	portEnv := []corev1.EnvVar{
+		{
+			Name:  userContainerDaprHTTPPortName,
+			Value: strconv.Itoa(sidecarHTTPPort),
+		},
+		{
+			Name:  userContainerDaprGRPCPortName,
+			Value: strconv.Itoa(sidecarAPIGRPCPort),
+		},
+	}
+	var value interface{}
+	for i, container := range list {
+		first := len(container.Env) == 0
+		for _, addEnv := range portEnv {
+			path := fmt.Sprintf("%s/%d/env", basePath, i)
```

Comment:
For readability, I would define new function and move the codes from line 170 to 193 to new function. 

e.g.
```
getEnvPatchOperation(envs *container.Envs, addVar coverv1.EnvVar) []PatchOperation {
line 170-193
}
```

then you can remove hasKey var and its if-statement and make this code cleaner and more readable.